File: Frontend/untitled/src/app/services/notification.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface NotificationDTO {
  id: string;
  title: string;
  message: string;
  type?: string;
  targetUrl?: string;
  read: boolean;
  createdAt: string;
}

@Injectable({ providedIn: 'root' })
export class NotificationService {
  private apiUrl = 'http://localhost:8080/api/notifications';
  constructor(private http: HttpClient) {}

  mine(): Observable<NotificationDTO[]> {
    return this.http.get<NotificationDTO[]>(this.apiUrl);
  }

  unreadCount(): Observable<{ count: number }> {
    return this.http.get<{ count: number }>(`${this.apiUrl}/unread-count`);
  }

  markRead(id: string): Observable<NotificationDTO> {
    return this.http.put<NotificationDTO>(`${this.apiUrl}/${id}/read`, {});
  }

  markAllRead(): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/read-all`, {});
  }
}

File: Backend/src/main/java/com/example/demo/service/NotificationService.java
package com.example.demo.service;

import com.example.demo.dto.NotificationDTO;
import com.example.demo.model.Person;

import java.util.List;

public interface NotificationService {
    NotificationDTO notifyUser(Person user, String title, String message, String type, String targetUrl);
    List<NotificationDTO> mine();
    long unreadCount();
    NotificationDTO markRead(String id);
    void markAllRead();
}
